### PR TITLE
[WIP] clang/tidy: Fixes (c++20 and cleanups)

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
mobile says:

```console
external/envoy/source/common/http/http2/codec_impl.h:429:32: error: default member initializer for bit-field is a C++20 extension [-Werror,-Wc++20-extensions]
```

**Review by adisuissa — COMMENTED** [review]
Small comment, but not sure if we can tweak the tool to address this

**Comment by phlax** [comment]
>  tweak the tool

config is in https://github.com/envoyproxy/envoy/blob/main/.clang-tidy

we can also `NOLINT` - but mho is that is the very last resort

File: contrib/qat/compression/qatzstd/compressor/source/config.h [hunk 1]
```diff
@@ -49,7 +49,7 @@ class QatzstdCompressorFactory : public Envoy::Compression::Compressor::Compress
                               public Logger::Loggable<Logger::Id::compression> {
     QatzstdThreadLocal();
     ~QatzstdThreadLocal() override;
-    void* GetQATSession();
+    void* getQATSession();
```
[the remark this answers is not in the document]
Comment (phlax) [reply contrib/qat/compression/qatzstd/compressor/source/config.h:52]: i can rerun again without fix to get err message ...

**Pull request closed without merging**